tests/python/unittest/test_higher_order_grad.py

```diff
@@ -173,6 +182,87 @@ def check_second_order_unary(x, op, grad_grad_op):
     # Validate the gradients.
     assert_almost_equal(expected_grad_grad, x.grad.asnumpy())
 
+def arange_shape_like(y):
+    shape = y.shape
+    nelems = reduce(mul, shape)
+    x = nd.arange(nelems).reshape(shape)
+    return x
+
+class RandomShapes(object):
```

Comment:
I feel this class is not necessary.

We already have:

```
shape = rand_shape_nd(dim)
```

Reply:
Is it such a big problem in a unit test? I find it useful to generate inputs.